libs/core/kiln_ai/datamodel/basemodel.py

```
@@ -71,6 +68,88 @@ def string_to_valid_name(name: str) -> str:
+
+    def copy_file_to(self, dest_folder: Path) -> Path:
+        filename = f"{str(uuid.uuid4().int)[:12]}{self.path.suffix}"
+        target_path = dest_folder / filename
+        shutil.copy(self.path, target_path)
+        return target_path
+
+    @classmethod
+    def from_data(
+        cls, data: str | bytes, mime_type: str, suffix: str | None = None
+    ) -> Self:
+        """Create an attachment from str or byte data, in a temp file. The attachment is persisted to
```

Pylint (3.3.7)

[convention] Line too long (101/100)

(C0301)